### ISSUE TITLE
Support Python 2.4 and 2.5 somehow

### ISSUE THREAD

**Issue by asmeurer** [sympy/sympy-bot-old#18]
Python 2.4 support isn't so important, since we are dropping support right after the release.  But 2.5 support is important.  The problem is that the json module doesn't exist there:

```
$python2.5 sympy-bot  --reference=~/Documents/python/sympy/sympy/ review 302
Traceback (most recent call last):
  File "sympy-bot", line 3, in <module>
    import json
ImportError: No module named json
```

But it should be possible to maybe run the tests a separate thread (with a separate Python).  We could also consider the possibility of running the tests on multiple Pythons in parallel.

**Comment by certik** [comment]
On Tue, May 10, 2011 at 9:34 PM, asmeurer
reply@reply.github.com
wrote:

> Python 2.4 support isn't so important, since we are dropping support right after the release.  But 2.5 support is important.  The problem is that the json module doesn't exist there:
> 
>    $python2.5 sympy-bot  --reference=~/Documents/python/sympy/sympy/ review 302
>    Traceback (most recent call last):
>      File "sympy-bot", line 3, in <module>
>        import json
>    ImportError: No module named json
> 
> But it should be possible to maybe run the tests a separate thread (with a separate Python).  We could also consider the possibility of running the tests on multiple Pythons in parallel.

Hm, I think that JSON is needed in order to communicate with github.
But I think that the json module is small and can be included with
sympy-bot, or rather simply require python2.6 for sympy-bot, and run a
different python for the tests.

Ondrej

**Comment by mattpap** [comment]
Use `simplejson` module. It has exactly the same API and functionality as `json` in >= Python 2.6.

**Comment by asmeurer** [comment]
Actually, even if you run sympy-bot with Python 2.6, it still runs tests in Python 2.7 (the default python).  This model is fine, though there should be a way to choose a different Python command to run the tests with.  If we do that, then Python 2.5 support doesn't matter so much (because all the devs who are using sympy-bot will have at least 2.6).

**Comment by asmeurer** [comment]
See issue #20.

**Comment by asmeurer** [comment]
I think it's just fine if sympy-bot and sympy-next themselves only support Python 2.7 (indeed, perhaps we should port them to Python 3).  So I'm closing this in favor of #20.
